extensions/sql-migration/src/wizard/skuRecommendationPage.ts

```diff
@@ -459,102 +449,91 @@ export class SKURecommendationPage extends MigrationWizardPage {
 			level: azdata.window.MessageLevel.Error
 		};
 
-		await this._setAssessmentState(true, false);
-
 		const serverName = (await this.migrationStateModel.getSourceConnectionProfile()).serverName;
-		const errors: string[] = [];
-		try {
-			if (this.migrationStateModel.resumeAssessment && this.migrationStateModel.savedInfo.closedPage) {
-				this.migrationStateModel._assessmentResults = <ServerAssessment>this.migrationStateModel.savedInfo.serverAssessment;
-			} else {
+
+		if (this.migrationStateModel._runAssessments) {
+			const errors: string[] = [];
+			await this._setAssessmentState(true, false);
+			try {
 				await this.migrationStateModel.getDatabaseAssessments(MigrationTargetType.SQLMI);
+				const assessmentError = this.migrationStateModel._assessmentResults?.assessmentError;
+				if (assessmentError) {
+					errors.push(`message: ${assessmentError.message}${EOL}stack: ${assessmentError.stack}`);
+				}
+				if (this.migrationStateModel?._assessmentResults?.errors?.length! > 0) {
+					errors.push(...this.migrationStateModel._assessmentResults?.errors?.map(
+						e => `message: ${e.message}${EOL}errorSummary: ${e.errorSummary}${EOL}possibleCauses: ${e.possibleCauses}${EOL}guidance: ${e.guidance}${EOL}errorId: ${e.errorId}`)!);
+				}
+			} catch (e) {
+				console.log(e);
+				errors.push(constants.SKU_RECOMMENDATION_ASSESSMENT_UNEXPECTED_ERROR(serverName, e));
```

Comment:
for all API call error handlers, can you add telemetry error logging here instead of console.log, it will send the telemetry error area and log to the console

Reply:
I addressed this particular one, but can we address the other ones in a separate pr? There's quite a few

Reply:
please fix where it applies to your changes and feel free to improve other areas as desired